File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
public class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int start = section[0];
        answer++;

        for (int item : section) {
            if (start + m > item){
                continue;
            }
            start = item;
            answer++;
        }

        return answer;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int n = 8;
        int m = 4;
        int[] section = {2, 3, 6};
        int answer = sol.solution(n, m, section);
        System.out.println("answer = " + answer);
    }
}

File: JavaScript/프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    var answer = 1;
	let index = section.pop();
	if (m == 1) {
		return section.length + 1;
	}

	while (section.length > 0) {
		let current = section.pop();

		if (current >= index - (m - 1)) {
			continue;
		}
		answer++;
		index = current;
	}

    return answer;
}